src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -521,6 +523,34 @@ function getDefaultConfig(): any {
   return JSON.parse(process.env.FIREBASE_CONFIG || "{}");
 }
 
+function initializeRuntimeConfig(frb: FunctionsRuntimeBundle) {
+  // Most recent version of Firebase Functions SDK automatically picks up locally
+  // stored .runtimeconfig.json to populate the config entries.
+  // However, due to a bug in some older version of the Function SDK, this process may fail.
+  // See the following issues for more detail:
+  //   https://github.com/firebase/firebase-tools/issues/3793
+  //   https://github.com/firebase/firebase-functions/issues/877
+  // As a workaround, the emulator runtime will load the contents of the .runtimeconfig.json
+  // to the CLOUD_RUNTIME_CONFIG environment variable.
+  // In the future, we will bump up the minimum version of the Firebase Functions SDK
+  // required to run the functions emulator to v3.15.1 and get rid of this workaround.
+  const configPath = `${frb.cwd}/.runtimeconfig.json`;
+  try {
+    const configContent = fs.readFileSync(configPath, "utf8");
+    if (configContent) {
+      try {
+        JSON.parse(configContent.toString());
+        logDebug(`Found local functions config: ${configPath}`);
+        process.env.CLOUD_RUNTIME_CONFIG = configContent.toString();
```

Comment:
I'm a bit unfamiliar with how this all works, so this might be off base - however, should we check if `process.env.CLOUD_RUNTIME_CONFIG` is already defined before we try to load it?

Reply:
Good catch. Updated.